source/frontend/src/components/maps/leaflet/Layers/PointClusterer.tsx

```diff
@@ -258,148 +215,142 @@ export const PointClusterer: React.FC<React.PropsWithChildren<PointClustererProp
-          const {
-            cluster: isCluster,
-            point_count: pointCount,
-            point_count_abbreviated,
-          } = cluster.properties as any;
-          const size = pointCount < 100 ? 'small' : pointCount < 1000 ? 'medium' : 'large';
-          // we have a cluster to render
-          if (isCluster) {
+
+          // Only clusters have the cluster property, if so we have a cluster to render
+          if ('cluster' in cluster.properties) {
+            const clusterFeature = cluster as ClusterFeature<ClusterProperties>;
+            const { point_count: pointCount, point_count_abbreviated } = clusterFeature.properties;
+
+            const sizeClass = pointCount < 100 ? 'small' : pointCount < 1000 ? 'medium' : 'large';
             return (
               // render the cluster marker
               <Marker
                 key={index}
                 position={[latitude, longitude]}
                 eventHandlers={{
                   click: e => {
-                    zoomOrSpiderfy(cluster);
+                    zoomOrSpiderfy(clusterFeature);
                     e.target.closePopup();
                   },
                 }}
                 icon={
                   new L.DivIcon({
                     html: `<div><span>${point_count_abbreviated}</span></div>`,
-                    className: `marker-cluster marker-cluster-${size}`,
+                    className: `marker-cluster marker-cluster-${sizeClass}`,
                     iconSize: [40, 40],
                   })
                 }
               />
             );
-          }
+          } else {
+            const clusterFeature = cluster as PointFeature<
+              PIMS_Property_Location_View | PMBC_FullyAttributed_Feature_Properties
+            >;
 
-          return (
-            // render single marker, not in a cluster
-            <Marker
-              {...(cluster.properties as any)}
-              key={index}
-              position={[latitude, longitude]}
-              icon={getMarkerIcon(cluster)}
-              eventHandlers={{
-                click: e => {
-                  const convertedProperty = convertToProperty(
-                    cluster.properties,
-                    latitude,
-                    longitude,
-                  );
-                  convertedProperty && onMarkerClick(convertedProperty); //open information slideout
-                  setCurrentSelected(convertedProperty);
-
-                  mapStateContext.setState({
-                    type: MapStateActionTypes.SELECTED_INVENTORY_PROPERTY,
-                    selectedInventoryProperty: convertedProperty,
-                  });
-                },
-              }}
-            />
-          );
+            const isSelected =
+              selectedFeature !== null ? clusterFeature.id === selectedFeature.clusterId : false;
+            // TODO: There might be a cleaner way to know if a marker is from a different source.
```

Comment:
wait, is it even possible to have clustered markers that aren't from PIMS?

Reply:
Yes, the clusters are coming from the feature points passed to the supercluster. So in theory it could have a mixture of both types of features. In practice it only has one type at a time